core/shared/src/main/scala/zio/Config.scala

```diff
@@ -0,0 +1,159 @@
+
+  def orElse[A1 >: A](that: Config[A1]): Config[A1] = self || that
+
+  def sequence: Config[Chunk[A]] = Config.Sequence(self)
+
+  def validate[A1 >: A](f: A1 => Boolean): Config[A1] =
+    self.mapOrFail(a =>
+      if (!f(a)) Left(Config.Error.Generic(Chunk.empty, "Validation of configuration data failed")) else Right(a)
+    )
+
+  def zip[B](that: Config[B])(implicit z: Zippable[A, B]): Config[z.Out] = self ++ that
+}
+object Config {
+  final case class Bool(name: String)                                                                  extends Config[Boolean]
+  final case class Constant[A](value: A)                                                               extends Config[A]
+  final case class Decimal(name: String)                                                               extends Config[BigDecimal]
+  final case class Duration(name: String)                                                              extends Config[java.time.Duration]
+  final case class Fail(message: String)                                                               extends Config[Nothing]
+  final case class Fallback[A](first: Config[A], second: Config[A])                                    extends Config[A]
+  final case class Integer(name: String)                                                               extends Config[BigInt]
+  final case class Labelled[A](config: Config[A], label: String)                                       extends Config[A]
+  final case class Lazy[A](thunk: () => Config[A])                                                     extends Config[A]
+  final case class LocalDateTime(name: String)                                                         extends Config[java.time.LocalDateTime]
+  final case class LocalDate(name: String)                                                             extends Config[java.time.LocalDate]
+  final case class LocalTime(name: String)                                                             extends Config[java.time.LocalTime]
+  final case class MapOrFail[A, B](original: Config[A], mapOrFail: A => Either[Config.Error, B])       extends Config[B]
+  final case class OffsetDateTime(name: String)                                                        extends Config[java.time.OffsetDateTime]
+  final case class Secret(name: String)                                                                extends Config[Chunk[Byte]]
+  final case class Sequence[A](config: Config[A])                                                      extends Config[Chunk[A]]
+  final case class Table[K, V](keyConfig: Config[K], valueConfig: Config[V])                           extends Config[Map[K, V]]
+  final case class Text(name: String)                                                                  extends Config[String]
+  final case class URL(name: String)                                                                   extends Config[java.net.URL]
```

Comment:
Isn't `java.net.URL` a [dangerous landmine](https://old.reddit.com/r/java/comments/qi8yu8/hint_to_myself_and_other_poor_souls_dont_use/)? Shouldn't this be rather `URI` (`java.net.URI`)?

Reply:
Fixed